modules/basegl/src/datastructures/meshshadercache.cpp

```diff
@@ -37,53 +37,48 @@ namespace inviwo {
 
 MeshShaderCache::Requirement::Requirement(BufferType bufferType, RequireBuffer required,
                                           const std::string& glslType, const std::string& name)
-    : bufferType{bufferType}
-    , required{required}
-    , glslType{glslType}
-    , name{name.empty() ? toString(bufferType) : name} {}
+    : getState{[bufferType, required](const Mesh& mesh, Mesh::MeshInfo) -> int {
+        const auto res = mesh.findBuffer(bufferType);
+        if (res.first) {
+            return res.second;
+        } else if (required == Mandatory) {
+            throw Exception("Unsupported mesh type, a " + toString(bufferType) + " is needed",
+                            IVW_CONTEXT_CUSTOM("MeshShaderCache"));
+        } else {
+            return -1;
+        }
+    }}
+    , updateShader{[glslType, name = name.empty() ? toString(bufferType) : name](
+                       int location, Shader& shader) -> void {
+        if (location >= 0) {
+            const auto& buffername = name;
+            shader[ShaderType::Vertex]->addInDeclaration("in_" + buffername, location, glslType);
+            for (auto& obj : shader.getShaderObjects()) {
+                obj.addShaderDefine("HAS_" + toUpper(buffername));
```

Comment:
Defines never removed?

Reply:
It only does that one when the shader is created, it the value is different a different shader is created. Hence it should never remove anything